monai/data/dataloader.py

```diff
@@ -80,6 +80,12 @@ def __init__(self, dataset: Dataset, num_workers: int = 0, **kwargs) -> None:
             init_seed = _g.initial_seed()
             _seed = torch.empty((), dtype=torch.int64).random_(generator=_g).item()
             set_rnd(dataset, int(_seed))
+            # disable unnecessary multiprocessing caching
+            from monai.data.dataset import CacheDataset  # avoid circular import
+
+            if isinstance(dataset, CacheDataset) and dataset.runtime_cache:
+                dataset._cache = dataset._set_multiprocessing_cache(dataset._cache, False)
+
```

Comment:
- another option is to use `hasattr(dataset, '_set_multiprocessing_cache')`, then we don't need to worry about cicular import

- `dataset._cache = dataset._set_multiprocessing_cache(dataset._cache, False)` this seems like a call to a private api (both `._set_multiprocessing_cache and ._cache` are private)
May be we can do it as `dataset.ensure_non_shared()`, which will ensure that cache is List and not ListProxy inside

Reply:
Sure, I simplified the API to avoid private functions and private variables in the latest commit.
But I didn't use the "hasattr" method, I think `if isinstance(dataset, CacheDataset)` is more robust.

Thanks.